src/conversation/prompts.rs

```diff
@@ -17,23 +17,10 @@ pub fn generate_completion_request(
     messages: Vec<ChatCompletionMessage>,
     function_call: FunctionCallType,
 ) -> ChatCompletionRequest {
-
-    ChatCompletionRequest {
-        model: CHAT_COMPLETION_MODEL.into(),
-        messages,
-        functions: Some(functions()),
-        function_call: Some(function_call),
-        temperature: Some(CHAT_COMPLETION_TEMPERATURE),
-        top_p: None,
-        n: None,
-        stream: None,
-        stop: None,
-        max_tokens: None,
-        presence_penalty: None,
-        frequency_penalty: None,
-        logit_bias: None,
-        user: None,
-    }
+    ChatCompletionRequest::new(CHAT_COMPLETION_MODEL.to_string(), messages)
```

Comment:
I know it's related to the version update, but can someone TLDr; the difference here?

Reply:
The library moved to a builder pattern.
Before we had to do something like this.
```rust
let req = ChatCompletionRequest {
    model: chat_completion::GPT4.to_string(),
    messages: vec![chat_completion::ChatCompletionMessage {
        role: chat_completion::MessageRole::user,
        content: String::from("What is Bitcoin?"),
        name: None,
        function_call: None,
    }],
    functions: None,
    function_call: None,
    temperature: 0.7,
    top_p: None,
    n: None,
    stream: None,
    stop: None,
    max_tokens: None,
    presence_penalty: None,
    frequency_penalty: None,
    logit_bias: None,
    user: None,
};
```
Now it's
```rust
let req = ChatCompletionRequest::new(
    chat_completion::GPT4.to_string(),
    vec![chat_completion::ChatCompletionMessage {
        role: chat_completion::MessageRole::user,
        content: String::from("What is Bitcoin?"),
        name: None,
        function_call: None,
    }],
).temperature(0.7);
```
Making it a lot less verbose and concise.

Reply:
We could _technically_ still use the types directly. But that's sort of a pain and like Anush said, it'll be way more concise and easy to read / maintain in the future.